### DISCUSSION TITLE
Importing of RDF into JanusGraph

### DISCUSSION THREAD

**Discussion by RituLalwani** [JanusGraph/janusgraph#2297] [Q&A]
Hi,

I want to know whether we can import RDF(Any format like Turtle or RDF/JSON) into Janusgraph or not.
If yes then what are the steps to do that.
Any help will highly be appreciated.

Thanks.

**Comment by porunov** [comment]
Hi @RituLalwani ,
It's definitely possible but I don't know if there are open source tools for that or not.
Searching very quickly I found a tool https://github.com/mpolonioli/rdf-to-janusgraph . I didn't check it and looks like it isn't supported but you can check their code which might help.
A related discussion can be found here: https://groups.google.com/g/janusgraph-users/c/I4rW_KhcgFE
Many companies are building their own RDF ingestion plugins around JanusGraph. I guess, soon we should see more and more tools to be open sourced.
You can check this whitepaper http://ceur-ws.org/Vol-2180/paper-30.pdf which talks about AI research at IBM. A phrase from the whitepaper:
> During ingestion, user datain various commonly used formats is converted and stored in the back-end storage. Inorder to ingest OWL/RDF, data we have extended RDF4J framework to add JanusGraphas additional store implementing data store interfaces. This takes RDF/OWL data invarious forms and maps it to appropriate property graph model and creates requiredindices as well.

Update:
Some resources to start with posted by Marc here https://groups.google.com/g/janusgraph-users/c/ATlv74gx188 :
https://tinkerpop.apache.org/docs/current/reference/#sparql-gremlin
https://pypi.org/project/rdf2gremlin/
https://docs.aws.amazon.com/neptune/latest/userguide/bulk-load-tutorial-format-rdf.html
